samples/subsys/usb/midi/README.rst

```diff
@@ -0,0 +1,81 @@
+.. zephyr:code-sample:: midi
+   :name: USB MIDI device class sample
+   :relevant-api: usbd_api usb_midi
+
+   USB MIDI 2.0 device class sample
+
+Overview
+********
+
+This sample demonstrates how to implement a USB MIDI 2.0 device. It can run on
+any board with a USB device controller. This samples sends all MIDI1 messages
+sent to the device back to the host. In addition, on boards that have a
+user button, pressing and releasing that button will send MIDI1 note on and
+note off messages.
+
+The application exposes a single USB-MIDI interface with a single bidirectional
+group terminal. This allows exchanging data with the host on a "virtual wire"
+that carries MIDI1 messages, pretty much like a standard USB-MIDI in/out adapter
+would provide. The loopback acts as if a real MIDI cable was connected between
+the output and the input, and the button acts as if there was a user keyboard
+to press a key.
+
+Building and Running
+********************
+
+The code can be found in :zephyr_file:`samples/subsys/usb/midi`.
+
+To build and flash the application:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/subsys/usb/midi
+   :board: nucleo_f429zi
+   :goals: build flash
+   :compact:
+
+Using the MIDI interface
+************************
+
+Once this sample is flashed, connect the device USB port to a host computer
+with MIDI support. For example, on Linux, you can use alsa to access the device:
+
+.. code-block:: console
+
+  $ amidi -l
+  Dir Device    Name
+  IO  hw:2,1,0  Group 1 (USBD MIDI Sample)
+
+
+The "USBD MIDI Sample" interface should also appear in any program with MIDI
+support (for example your favorite Digital Audio Workstation)
```

Comment:
maybe also mention some of the web based tools that exist, as this makes things even easier to test. We could *almost* have a minimal hosted html/js page similar to https://docs.zephyrproject.org/latest/samples/subsys/usb/webusb/demo.html in the tree to make it dead easy for folks to test / play with this.

Reply:
I wasn't familiar with webusb so I looked into that, and I discovered that there's even a webmidi API. I would rather find a demo based on the latter, otherwise we'd have to re-build usb-midi host support in the demo :stuck_out_tongue: 

By the way, is your video based on this website https://muted.io/piano/ ? Maybe we could directly refer to that ?